src/posix/platform/include/openthread/openthread-system.h

```diff
@@ -330,6 +330,14 @@ void otSysUpstreamDnsServerSetResolvConfEnabled(bool aEnabled);
  */
 void otSysUpstreamDnsSetServerList(const otIp6Address *aUpstreamDnsServers, int aNumServers);
 
+/**
+ * Sets the radio url to configure the radio related parameters.
+ *
+ * @param[in] aRadioUrl  A pointer to the radio url string.
+ *
+ */
+void otSysSetRadioUrl(const char *aRadioUrl);
```

Comment:
I noticed that within `Radio::Init(const char *aUrl)` there are cetain checks that are peformed before calling `    ProcessRadioUrl(mRadioUrl);`, in paricular:
```
    resetRadio             = !mRadioUrl.HasParam("no-reset");
    skipCompatibilityCheck = mRadioUrl.HasParam("skip-rcp-compatibility-check");
```
Also have this check
```
    mRadioUrl.Init(aUrl);
    VerifyOrDie(mRadioUrl.GetPath() != nullptr, OT_EXIT_INVALID_ARGUMENTS);
```

How is this new API which allows setting the URL later, intended to work along with these checks from `Radio::Init()`?

Reply:
Having the same question. Is `otSysSetRadioUrl` called earlier than Radio init somehow?

Reply:
Current architecture of the `radio.cpp` and `system.cpp` initiates the radio URL when the OT host starts. Some of the parameters, such as  `skip-rcp-compatibility-check`, must be initialized when the Thread host starts. But the parameters, such as `cca-threshold`, can be updated after the Thread host starts. If we are looking at the `ProcessRadioUrl()`, we will find that all parameters in this function can be updated after the Thread host starts.

Current function `otSysSetRadioUrl ` only set the parameters that can be updated after thread host starts. We can allow the `otSysSetRadioUrl` to change all parameters after we changed the architecture of the `radio.cpp` and `system.cpp` to allow lazy initialization. Thoughts?